### PR TITLE
contrib/systemd cleanups

### PR DESCRIPTION
**Pull request by vrothberg**
This PR includes a series of fixes and clean ups for the systemd service and socket. Please refer to the individual commits for details.

Fixes: #7021

### CODE REVIEW

**Comment by vrothberg** [comment]
@giuseppe @jwhonce @mheon  @rhatdan PTAL

File: contrib/systemd/system/podman.service [hunk 1]
```diff
@@ -1,7 +1,8 @@
 [Unit]
 Description=Podman API Service
 Requires=podman.socket
-After=podman.socket
+Wants=network.target
+After=podman.socket network-online.target
```
Comment (martinpitt) [inline contrib/systemd/system/podman.service:5]: As former systemd developer/maintainer I recommend to not do that. podman.service does *not* need networking at startup, only for some operations -- but that shouldn't affect the boot order. network-online.target is a rather terrible and brittle hack, see `man systemd.special`.

`Wants=network.target` is also unnecessary here. Your unit shouldn't activate that -- at most you'd have `After=network.target`, but that's not even required as podman does not require the network to shut down cleanly.

**Comment by vrothberg** [comment]
Thanks for reviewing and the feedback, @martinpitt. Your comments are addressed in the latest state.

**Comment by rhatdan** [comment]
@goochjj PTAL

**Comment by TomSweeneyRedHat** [comment]
LGTM

**Comment by goochjj** [comment]
LGTM

**Pull request merged**